src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Properties.jl

```diff
@@ -17,7 +17,7 @@ end
 end
 
 @doc raw"""
-    is_smooth(P::AbsProjectiveScheme; algorithm=:default) -> Bool
+    is_smooth(P::AbsProjectiveScheme{<:Any, <:MPolyQuoRing}; algorithm=:default) -> Bool
 
```

Comment:
We could rather catch the easy corner case than exclude it in the docs. 
It is better, if we have the compatibility functions to do this in one place (what you do below anyway), than several users doing it each by themselves after reading this in the docs.

Reply:
A solution could be the following: Define

`is_smooth(P::AbsProjectiveScheme; algorithm=:default)`
and document just this method. Then have the following two methods
`is_smooth(P::AbsProjectiveScheme{<:any, <:MPolyQuoRing}; algorithm=:default)`
`is_smooth(P::AbsProjectiveScheme{<:any, <:MPolyRing}; algorithm=:default)`

perhaps the `any` needs to be replaced by `Field` ? ... I have little experience with algebraic geometry over general rings ... best consult a textbook to be sure.

Reply:
We already have is_smooth for MPolyRing below in the new code. This is the way to go

Over general rings, smooth means Jacobi-criterion satisfied and regular means dim = edim. Please keep in mind that regular still allows fibre-singularities e.g. over ZZ.